packages/material-ui-lab/src/SliderStyled/SliderStyled.d.ts

```diff
@@ -48,4 +48,46 @@ export const SliderValueLabel: React.FC<SliderValueLabel>;
  */
 declare const Slider: ExtendSliderUnstyled<SliderStyledTypeMap>;
 
+export type SliderStyledClassKey =
```

Comment:
How would this type be used?

Reply:
We do not have the `classes` prop anymore on this component, so I was thinking that we can use this type for generating the `CSS` section on the docs (the type is also useful for the theme overrides keys. You can check out the changes on the `slider-styled.md`

Reply:
If we want to generate documentation then why use types rather than e.g. basic JSON?

Reply:
Should we then add it on the static options object on the component rather than here? We will anyway need the typing, that’s why I decided to use that one.

Reply:
>  We will anyway need the typing, that’s why I decided to use that one.

That was my original question: Where do we need it? How do we use it?

> Should we then add it on the static options object on the component rather than here?

If it's not needed at runtime we shouldn't bloat the bundle. If the types aren't used they shouldn't be added as types.

I'm missing some context here. The PR only talks about documentation so any component runtime/types shouldn't be touched in the first place.

Reply:
Ah, sorry for the confusion. All components currently export `ComponentNameClassKey` type which is used for typing the overrides inside the theme in https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/styles/overrides.d.ts Currently this type is just re-exporting an union type from the `classes` prop, and the same `classes` prop is used for the documentation. However, we are dropping the `classes` prop from the components, so the only source of truth for these values is this `type`. That's why I've added this additional logic for extracting it from the types if the `classes` prop doesn't exist.

Reply:
So it would make more sense to have it typed out like `classes` in `overrides` (or rather with module augmentation right now) and extract it from there? 

If I understood it correctly we don't need these `*ClassKey` types anymore in the future so now is  a good time to get rid of them.

Reply:
If I understand you correctly, yes we could have it defined directly in the `overrides.d.ts` and if that is the case we won't need it anymore as a type defined in the `Component.d.ts` file anymore. Should we do that? I am ok with any of the two options. However for the lab we would need to extract it from https://github.com/mui-org/material-ui/blob/next/packages/material-ui-lab/src/themeAugmentation/overrides.d.ts

Reply:
> Should we do that?

If we don't need to don't do it. It's as simple as that. The ClassKey types were a convenience helper but since it was decided to drop that API we might as well leverage that and get rid of more code.

> However for the lab we would need to extract it from

Let's worry about this in another PR. One step at a time. Right now the lab does not include SliderStyled overrides.

Reply:
I would keep this PR minimal to just add the functionality for extracting the info from the types, we will anyway need it regardless of whether the type lives in the `Component.d.ts` file or the `overrides.d.ts` file. If we decide to remove these typings for sure, we can do it in a follow up PR that would do this for all components and update the script to extract the info from the `overrides.d.ts`. Is that ok?